### PR TITLE
Updated Errata entity and added heplers for errata/packages installation

### PR DESCRIPTION
**Pull request by abalakh**
* Added 13 new fields for Errata entity
* Added following helpers: `Errata().compare()`, `Host().errata()`,
`Host().errata_apply()`, `Host().install_content()`,
`Repository().errata()`.
* Added unit tests for new helpers

### CODE REVIEW

**Comment by abalakh** [comment]
```python
In [1]: errata = entities.Errata(id='baeb6fd5-b044-4dab-8ada-870692dc2a97').read()

In [2]: errata.description
Out[2]: u'Red Hat Satellite is a systems management tool for Linux-based infrastructures.\nIt allows for provisioning, remote management and monitoring of multiple Linux\ndeployments with a single, centralized tool. It performs provisioning and\nconfiguration management of predefined standard operating environments.\n\nThis update provides Satellite 6.2 client tooling. For the full list of new\nfeatures provided by Satellite 6.2 see the Release Notes linked to in the\nreferences section. Please see the Satellite 6 Installation Guide for detailed\ninstructions on how to install or upgrade from prior versions of Satellite 6.\n\nAll users who require Satellite version 6.2 are advised to install these new\npackages.'

In [3]: errata.hosts_applicable_count
Out[3]: 0

In [4]: errata.hosts_available_count
Out[4]: 0

In [5]: errata.issued
Out[5]: u'2016-07-27'

In [6]: errata.packages
Out[6]: 
[u'facter-2.4.6-3.el7sat.x86_64',
 u'gofer-2.7.6-1.el7sat.noarch',
 u'hiera-1.3.1-2.el7sat.noarch',
 u'katello-agent-2.5.0-5.el7sat.noarch',
 u'libtasn1-3.8-3.el7.x86_64',
 u'libtasn1-devel-3.8-3.el7.x86_64',
 u'pulp-puppet-tools-2.8.3.3-1.el7sat.noarch',
 u'pulp-rpm-handlers-2.8.3.5-1.el7sat.noarch',
 u'puppet-3.8.6-2.el7sat.noarch',
 u'python-gofer-2.7.6-1.el7sat.noarch',
 u'python-gofer-proton-2.7.6-1.el7sat.noarch',
 u'python-isodate-0.5.0-4.el7sat.noarch',
 u'python-pulp-agent-lib-2.8.3.3-1.el7sat.noarch',
 u'python-pulp-common-2.8.3.3-1.el7sat.noarch',
 u'python-pulp-puppet-common-2.8.3.3-1.el7sat.noarch',
 u'python-pulp-rpm-common-2.8.3.5-1.el7sat.noarch',
 u'python-qpid-proton-0.9-16.el7.x86_64',
 u'qpid-proton-c-0.9-16.el7.x86_64',
 u'ruby-augeas-0.5.0-1.el7.x86_64',
 u'ruby-rgen-0.6.5-2.el7sat.noarch',
 u'ruby-shadow-1.4.1-21.el7.x86_64',
 u'rubygem-foreman_scap_client-0.1.2-1.el7sat.noarch',
 u'satellite-cli-6.2.0-21.1.el7sat.noarch',
 u'tfm-rubygem-ansi-1.4.3-6.el7sat.noarch',
 u'tfm-rubygem-apipie-bindings-0.0.16-1.el7sat.noarch',
 u'tfm-rubygem-awesome_print-1.0.2-13.el7sat.noarch',
 u'tfm-rubygem-clamp-1.0.0-4.el7sat.noarch',
 u'tfm-rubygem-fast_gettext-0.9.2-3.el7sat.noarch',
 u'tfm-rubygem-hammer_cli-0.5.1.11-4.el7sat.noarch',
 u'tfm-rubygem-hammer_cli_csv-2.0.0.1-1.el7sat.noarch',
 u'tfm-rubygem-hammer_cli_foreman-0.5.1.9-1.el7sat.noarch',
 u'tfm-rubygem-hammer_cli_foreman_bootdisk-0.1.3.3-1.el7sat.noarch',
 u'tfm-rubygem-hammer_cli_foreman_discovery-0.0.2.3-1.el7sat.noarch',
 u'tfm-rubygem-hammer_cli_foreman_docker-0.0.6-1.el7sat.noarch',
 u'tfm-rubygem-hammer_cli_foreman_remote_execution-0.0.5.3-1.el7sat.noarch',
 u'tfm-rubygem-hammer_cli_foreman_tasks-0.0.10.3-1.el7sat.noarch',
 u'tfm-rubygem-hammer_cli_import-0.11.2-1.el7sat.noarch',
 u'tfm-rubygem-hammer_cli_katello-0.0.22.23-1.el7sat.noarch',
 u'tfm-rubygem-hashie-2.0.5-4.el7sat.noarch',
 u'tfm-rubygem-highline-1.6.21-4.el7sat.noarch',
 u'tfm-rubygem-little-plugger-1.1.3-20.el7sat.noarch',
 u'tfm-rubygem-locale-2.0.9-11.el7sat.noarch',
 u'tfm-rubygem-logging-1.8.2-3.el7sat.noarch',
 u'tfm-rubygem-multi_json-1.10.1-3.el7sat.noarch',
 u'tfm-rubygem-netrc-0.7.7-7.el7sat.noarch',
 u'tfm-rubygem-oauth-0.4.7-8.el7sat.noarch',
 u'tfm-rubygem-powerbar-1.0.11-11.el7sat.noarch',
 u'tfm-rubygem-rest-client-1.6.7-4.el7sat.noarch',
 u'tfm-rubygem-table_print-1.5.1-5.el7sat.noarch',
 u'tfm-runtime-2.0-2.el7sat.x86_64']

In [7]: errata.reboot_suggested
Out[7]: False

In [8]: errata.severity
Out[8]: u'None'

In [9]: errata.solution
Out[9]: u'Before applying this update, make sure all previously released errata\nrelevant to your system have been applied.\n\nFor details on how to apply this update, refer to:\n\nhttps://access.redhat.com/articles/11258'

In [10]: errata.summary
Out[10]: u'Red Hat Satellite 6.2 now available for Red Hat Enterprise Linux 6 and 7'

In [11]: errata.type
Out[11]: u'bugfix'

In [12]: errata.updated
Out[12]: u'2016-07-27'
```

**Comment by oshtaier** [comment]
ACK pending also examples for added helpers

File: nailgun/entities.py [hunk 1]
```diff
@@ -2535,6 +2593,57 @@ def create_payload(self):
...
+            ``True``. Immediately return the server's response otherwise.
+        :param kwargs: Arguments to pass to requests.
+        :returns: The server's response, with all content decoded.
+        :raises: ``requests.exceptions.HTTPError`` If the server responds with
+            an HTTP 4XX or 5XX message.
+
+        """
+        kwargs = kwargs.copy()  # shadow the passed-in kwargs
+        kwargs.update(self._server_config.get_client_kwargs())
+        response = client.get(self.path('errata'), **kwargs)
```
Comment (omaciel) [inline nailgun/entities.py]: Do we need to document this path in the `path` method?

File: nailgun/entities.py [hunk 2]
```diff
@@ -2535,6 +2593,57 @@ def create_payload(self):
...
+            ``True``. Immediately return the server's response otherwise.
+        :param kwargs: Arguments to pass to requests.
+        :returns: The server's response, with all content decoded.
+        :raises: ``requests.exceptions.HTTPError`` If the server responds with
+            an HTTP 4XX or 5XX message.
+
+        """
+        kwargs = kwargs.copy()  # shadow the passed-in kwargs
+        kwargs.update(self._server_config.get_client_kwargs())
+        response = client.put(self.path('errata/apply'), **kwargs)
```
Comment (omaciel) [inline nailgun/entities.py]: Do we need to document this path in the `path` method?

File: nailgun/entities.py [hunk 3]
```diff
@@ -2577,6 +2686,12 @@ def path(self, which=None):
         """Extend ``nailgun.entity_mixins.Entity.path``.
         The format of the returned path depends on the value of ``which``:
 
+        bulk/install_content
```
Comment (omaciel) [inline nailgun/entities.py]: Is this path supported by a different `path` method?

**Comment by abalakh** [comment]
@oshtaier all these helpers are widely used in SatelliteQE/robottelo#3829, please take a look at test results report there.
Basically fetching, installing and especially comparing errata will require a lot of pre-conditions so i'm not sure if it's worth proving helpers are working manually, easier to see them in action in tests.
Please let me know if that's ok for you.

**Comment by sghai** [comment]
ACK.. I'm leaving this PR for @oshtaier if all your comments are addressed then lets merge it.

**Pull request merged**